File: backend/src/main/java/com/bupt/publicopinion/collection/vo/TopicSearchResult.java
package com.bupt.publicopinion.collection.vo;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record TopicSearchResult(
        String keyword,
        @JsonProperty("source_name") String sourceName,
        @JsonProperty("source_type") String sourceType,
        @JsonProperty("source_url") String sourceUrl,
        @JsonProperty("total_found") int totalFound,
        @JsonProperty("total_success") int totalSuccess,
        @JsonProperty("total_failed") int totalFailed,
        List<NewsCrawlResult> articles,
        List<FailedNewsCrawl> failures
) {
}

File: services/crawler/src/model/topic_search_result.py
from pydantic import BaseModel, Field

from src.model.crawl_result import CrawlResult
from src.model.news_collect_result import FailedNewsCrawl


class TopicSearchResult(BaseModel):
    keyword: str
    source_name: str
    source_type: str
    source_url: str
    total_found: int
    total_success: int
    total_failed: int
    articles: list[CrawlResult] = Field(default_factory=list)
    failures: list[FailedNewsCrawl] = Field(default_factory=list)
